### PR TITLE
feat: use GitHub Codespaces

### PR DESCRIPTION
**Pull request by lloydchang**
Co-authored-by: Haley Lifrieri <haleylifrieri@college.harvard.edu>
Co-authored-by: Haley Lifrieri <halifrieri@gmail.com>
Co-authored-by: Muturi David <muturidavid854@gmail.com>
Co-authored-by: Pardaz Banu Mohammad <pardaz.banu786@gmail.com>
Co-authored-by: Pardaz Banu Mohammad <pardazbanu1999@gmail.com>
Co-authored-by: Toluwalope Olateru-Olagbeg <wole2003@gmail.com>

### CODE REVIEW

**Comment by lloydchang** [comment]
Relates to https://github.com/ipa-lab/hackingBuddyGPT/issues/36
and https://github.com/ipa-lab/docs.hackingbuddy/pull/5
plus https://github.com/ipa-lab/hackingBuddyGPT/pull/85
then https://github.com/ipa-lab/benchmark-privesc-linux/pull/1

**Comment by lloydchang** [comment]
@andreashappe This https://github.com/ipa-lab/benchmark-privesc-linux/pull/2 didn't auto-assign to you for review.
Thank you!

**Comment by lloydchang** [comment]
The attached video `HackingBuddyGPT-and-GitHub-Codespaces.mp4` demonstrates the usage of this pull request via https://github.com/lloydchang/ipa-lab-benchmark-privesc-linux/tree/lloydchang-patch-4

I'm currently experiencing rate-limiting issues from OpenAI, preventing me from running more tests. However, that is a separate topic.

@andreashappe, thank you!

https://github.com/user-attachments/assets/56677b7a-4395-43dc-bc26-7149fa5f50a5

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+RUN wget https://bootstrap.pypa.io/get-pip.py && python3.11 get-pip.py && rm get-pip.py
+
+# Install passlib using pip
+RUN python3.11 -m pip install passlib
+
+# Prepare SSH server
+RUN mkdir /var/run/sshd
+
+# Set root password
+RUN echo "root:aim8Du7h" | chpasswd
```
Comment (andreashappe) [inline Dockerfile]: is using a hard-coded password problematic? or are the container only accessible from the current workspace (for the current user)?

Comment (lloydchang) [reply Dockerfile]: If the Docker host exposes dynamic ports to other IP addresses in the local area network, it creates a security risk.

An upcoming Git commit will address this problem:
> fix: security
> * Remove hardcoded root password
> * Prevent root SSH access
> * Use privilege escalation to root

Comment (lloydchang) [reply Dockerfile]: Added https://github.com/ipa-lab/benchmark-privesc-linux/pull/2/commits/61ba0ced8a7e5defb5f857861a099c3efb3a4c89 to this pull request.

**Comment by lloydchang** [comment]
This pull request starts multiple containers in Codespaces. Your review is appreciated.

It simulates `create_and_start_vms.sh` functionality as follows:
1. Revised `README.md`
2. `.devcontainer/devcontainer.json` follows official specs at https://devcontainers.github.io/implementors/json_reference/
3. `codespaces_create_and_start_containers.Dockerfile` restricts root SSH access
4. `codespaces_create_and_start_containers.sh` leverages `hosts.ini` and `tasks.yaml`

@andreashappe, thank you!

**Pull request merged**